ext-src/services/IqRequestService.ts

```diff
@@ -58,20 +59,28 @@ export class IqRequestService implements RequestService {
           url: `${this.url}/api/v2/applications?publicId=${applicationPublicId}`,
           headers: RequestHelpers.getUserAgentHeader(),
           auth: { user: this.user, pass: this.password }
-        },
-        (err: any, response: any, body: any) => {
-          if (err) {            
-            reject(`Unable to retrieve Application ID: ${err}`);
+        })
+        .on('response', (res) => {
+          console.log(res.statusCode);
+
+          if (res.statusCode != HttpStatus.OK) {            
+            reject(`Unable to retrieve Application ID. Could not communicate with server. Server error: ${res.statusCode}`);
             return;
           }
-          if (response.statusCode != HttpStatus.OK) {            
-            reject(`Unable to retrieve Application ID. Could not communicate with server. Server error: ${response.statusCode}`);
+
+          res.on('data', (data) => {
+            resolve(data);
+            return;
+          });
+        }).on('error', (err) => {
+          console.debug(err);
+          if (err.message.includes('ECONNREFUSED')) {
```

Comment:
Had to do this ugly hack because the err `Error` type doesn't have `code` on it at compile time, but does at runtime. 🤷‍♂